### PR TITLE
Categories filter

### PR DESCRIPTION
**Pull request by bhtibrewal**
- added categories
- 404 page
- theme

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -6,24 +6,30 @@ import {
...
-        <Route path="/quiz/:quizTitle" element={<QuestionPage />} />
+        <Route path="/quizzes/:categoryName" element={<Quizzes />} />
         <Route path="/quiz/rules" element={<RulesPage />} />
+        <Route path="/quiz/:quizTitle" element={<QuestionPage />} />
         <Route path="/result" element={<ResultPage />} />
         <Route path="/sign-in" element={<SignIn />} />
         <Route path="/sign-up" element={<SignUp />} />
+        <Route path="*" element={<PageNotFound />} />
       </Routes>
       {/* <Footer /> */}
```
Comment (pankaj-prog) [inline src/App.jsx]: remove commented code

File: src/components/header/Header.jsx [hunk 2]
```diff
@@ -22,7 +21,7 @@ export const Header = () => {
             <FaUser />
           </Link>
         )}
-        <span>
+        <span onClick={() => setDarkMode((prev) => !prev)}>
           <FaCloudMoon />
         </span>
```
Comment (pankaj-prog) [inline src/components/header/Header.jsx]: use button for actions

File: src/components/toast/Toast.jsx [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-import { useToast } from "../../contexts";
+import { useToast } from "contexts";
 import { useEffect } from "react";
 import {
   FaInfoCircle,
   FaExclamationCircle,
   FaCheckCircle,
-  IoCloseCircleOutline
+  IoCloseCircleOutline,
 } from "../../assets/icons";
```
Comment (pankaj-prog) [inline src/components/toast/Toast.jsx]: you can use absolute import for this as well

File: src/pages/page_not_found/PageNotFound.jsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import "./page_not_found.css";
+import image from "../../assets/images/404_image.svg";
+// import image from '../../assets/images/'
```
Comment (pankaj-prog) [inline src/pages/page_not_found/PageNotFound.jsx]: remove commented code

**Review by pankaj-prog — APPROVED** [review]

File: src/pages/page_not_found/PageNotFound.jsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+export const PageNotFound = () => {
+    return (
+      <main className="main center">
+        <img className="img_404" src={image} />
```
Comment (gauravsinhaweb) [inline src/pages/page_not_found/PageNotFound.jsx]: `alt` is missing

**Review by gauravsinhaweb — APPROVED** [review]

**Review by SinchanaMS — APPROVED** [review]
I couldn't find any issues other than the ones already mentioned by others. Please look into them and you're good to go!

**Pull request merged**